local-cluster/bench/Main.hs

```diff
@@ -7,23 +7,59 @@ import Bench.EndToEnd (bench)
 import Data.Aeson (eitherDecodeFileStrict', encodeFile)
 import Hydra.Ledger (applyTransactions)
 import Hydra.Ledger.Cardano (cardanoLedger, genSequenceOfValidTransactions, genUtxo)
+import Options.Applicative (Parser, ParserInfo, auto, execParser, fullDesc, header, help, helper, info, long, metavar, option, progDesc, strOption, value)
 import System.Environment (withArgs)
 import System.FilePath ((</>))
 import Test.QuickCheck (generate, scale)
 
+data Options = Options
+  { outputDirectory :: Maybe FilePath
+  , scalingFactor :: Int
+  }
+
+benchOptionsParser :: Parser Options
+benchOptionsParser =
+  Options
+    <$> optional
+      ( strOption
+          ( long "output-directory"
+              <> help "Directory containing generated transactions and UTxO set to use for replaying a previous benchmark."
+          )
+      )
+      <*> option
+        auto
+        ( long "scaling-factor"
+            <> value 100
+            <> metavar "INT"
+            <> help "The scaling factor to apply to transactions generator (default: 100)"
+        )
+
+benchOptions :: ParserInfo Options
+benchOptions =
+  info
+    (benchOptionsParser <**> helper)
+    ( fullDesc
+        <> progDesc
+          "Starts a cluster of Hydra nodes interconnected through a network and \
+          \ talking to mock-chain, generates an initial UTxO set and a bunch \
+          \ of valid transactions, and send those transactions to the cluster as \
+          \ fast as possible."
+        <> header "bench - load tester for Hydra node cluster"
+    )
+
 main :: IO ()
 main =
-  getArgs >>= \case
-    [benchDir] -> do
+  execParser benchOptions >>= \case
+    Options{outputDirectory = Just benchDir} -> do
       txs <- either die pure =<< eitherDecodeFileStrict' (benchDir </> "txs.json")
       utxo <- either die pure =<< eitherDecodeFileStrict' (benchDir </> "utxo.json")
       putStrLn $ "Using UTxO and Transactions from: " <> benchDir
       run benchDir utxo txs
-    _ -> do
+    Options{scalingFactor} -> do
```

Comment:
Slightly confusing, but this pattern match suggests that we ignore scalingFactor when a benchDir is set. Feels like options should really be a sum-type no ?

Reply:
Actually, in a later commit, I plan to use both so that we can _set_ the output directory where to generate transactions and store results

Reply:
This is useful for example if you want to run a benchmark with more load than your tmpfs can stand